db/appsock_handler.c

```diff
@@ -265,6 +268,33 @@ static void appsock_work(struct thdpool *pool, void *work, void *thddata)
     struct appsock_thd_state *state = thddata;
     appsock_work_args_t *w = (appsock_work_args_t *)work;
     int keepsocket = 0;
+    int curr_appsock_conns = 0;
+    w->rejected = 0;
+    int num_retry = 0;
+
+retry:
+    Pthread_mutex_lock(&appsock_conn_lk);
+    if (active_appsock_conns >= bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) {
+        if (close_lru_client(active_appsock_conns, NULL, bdb_attr_get(thedb->bdb_attr, BDB_ATTR_MAXAPPSOCKSLIMIT)) != 0) {
+            num_retry++;
+            if (num_retry < 5) {
+                Pthread_mutex_unlock(&appsock_conn_lk);
+                sleep(1);
```

Comment:
Maybe we should just reject and not retry?  The API will try on a different node, which is likely to be faster than waiting here.  The API already has logic to back off and retry.

Reply:
We don't want client to keep trying and thus increasing the load, we have to give error back.